samcli/lib/build/app_builder.py

```diff
@@ -819,6 +820,8 @@ def _build_function_on_container(
                     raise UnsupportedBuilderLibraryVersionError(
                         container.image, "{} executable not found in container".format(container.executable_name)
                     ) from ex
+            except ContainerStartTimeoutException as e:
+                raise BuildInsideContainerError(str(e)) from e
```

Comment:
I did this because `BuildInsideContainerError` is caught and re-raised as a UserException here: https://github.com/aws/aws-sam-cli/blob/develop/samcli/commands/build/build_context.py#L237-L251 .
`ContainerStartTimeoutException` wouldn't be caught and re-raised as a UserException otherwise and it would show the whole stack trace in a not very user friendly way. 

Is the preferred way to do it like this or to catch `ContainerStartTimeoutException` and re-raise as UserException as in the link above ^?